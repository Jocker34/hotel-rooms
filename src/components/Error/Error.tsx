import { Title, Text, Container, Group } from '@mantine/core';
import classes from './Error.module.css';

export const Error = () => (
  <Container className={classes.root}>
    <div className={classes.label}>Error</div>
    <Title className={classes.title}>Something went wrong.</Title>
    <Text c="dimmed" size="lg" ta="center" className={classes.description}>
      Please contact your administrator or try refreshing the page.
    </Text>
    <Group justify="center"></Group>
  </Container>
);
