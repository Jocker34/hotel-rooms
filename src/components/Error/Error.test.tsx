import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import { Error } from './Error';

describe('Error component', () => {
  test('renders error label', () => {
    render(<Error />);
    const labelElement = screen.getByText(/Error/i);
    expect(labelElement).toBeInTheDocument();
  });

  test('renders error title', () => {
    render(<Error />);
    const titleElement = screen.getByText(/Something went wrong/i);
    expect(titleElement).toBeInTheDocument();
  });

  test('renders error description', () => {
    render(<Error />);
    const descriptionElement = screen.getByText(/Please contact your administrator or try refreshing the page/i);
    expect(descriptionElement).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(<Error />);
    expect(asFragment()).toMatchSnapshot();
  });
});
