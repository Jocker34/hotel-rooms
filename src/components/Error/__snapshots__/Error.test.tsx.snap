// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error component > matches snapshot 1`] = `
<DocumentFragment>
  <style
    data-mantine-styles="classes"
  >
    @media (max-width: 35.99375em) {.mantine-visible-from-xs {display: none !important;}}@media (min-width: 36em) {.mantine-hidden-from-xs {display: none !important;}}@media (max-width: 47.99375em) {.mantine-visible-from-sm {display: none !important;}}@media (min-width: 48em) {.mantine-hidden-from-sm {display: none !important;}}@media (max-width: 61.99375em) {.mantine-visible-from-md {display: none !important;}}@media (min-width: 62em) {.mantine-hidden-from-md {display: none !important;}}@media (max-width: 74.99375em) {.mantine-visible-from-lg {display: none !important;}}@media (min-width: 75em) {.mantine-hidden-from-lg {display: none !important;}}@media (max-width: 87.99375em) {.mantine-visible-from-xl {display: none !important;}}@media (min-width: 88em) {.mantine-hidden-from-xl {display: none !important;}}
  </style>
  <div
    class="_root_f2d29a m_7485cace mantine-Container-root"
  >
    <div
      class="_label_f2d29a"
    >
      Error
    </div>
    <h1
      class="_title_f2d29a m_8a5d1357 mantine-Title-root"
      data-order="1"
      style="--title-fw: var(--mantine-h1-font-weight); --title-lh: var(--mantine-h1-line-height); --title-fz: var(--mantine-h1-font-size);"
    >
      Something went wrong.
    </h1>
    <p
      class="mantine-focus-auto _description_f2d29a m_b6d8b162 mantine-Text-root"
      data-size="lg"
      style="--text-fz: var(--mantine-font-size-lg); --text-lh: var(--mantine-line-height-lg); text-align: center;"
    >
      Please contact your administrator or try refreshing the page.
    </p>
    <div
      class="m_4081bf90 mantine-Group-root"
      style="--group-gap: var(--mantine-spacing-md); --group-align: center; --group-justify: center; --group-wrap: wrap;"
    />
  </div>
</DocumentFragment>
`;
